client/consensus/grandpa/authorities.go

```diff
@@ -725,27 +725,28 @@ func (asc *authoritySetChangeID) Value() (val scale.VaryingDataTypeValue, err er
 }
 
 func newAuthoritySetChangeID[N constraints.Unsigned]() authoritySetChangeID {
-	vdt := scale.MustNewVaryingDataType(latest{}, set[N]{}, unknown{})
+	vdt := scale.MustNewVaryingDataType(authoritySetChangeIDLatest{}, authoritySetChangeIDSet[N]{},
```

Comment:
Since those types don't need to be scale encoded/decoded I would suggest that you use interfaces to represent enums, using interfaces brings advantages like, 1) not using reflection overhead, 2) is error prune and you will be bound since the concrete type will be bound to the interface

```go
type AuthoritySetChange interface {
    isAuthoritySetChange()
}


type Latest struct {}
func (Latest) isAuthoritySetChange() {}

type Set[N constraints.Unsigned] struct {
    inner setIdNumber[N]
}
func (Set) isAuthoritySetChange() {}

type Unknown struct {}
func (Unknown) isAuthoritySetChange() {}
```

and instead of having `authSetChangeID := newAuthoritySetChangeID[N]()` you just create a variable like this `var authSetChangeID AuthoritySetChange`, and instead of calling the `.Set(...)` and checking the error you can just `authSetChangeID = Latest{}`, `authSetChangeID = Set{inner: ...}` or `authSetChangeID = Unknown{}`